src/pz390.java

```diff
@@ -13875,14 +13936,28 @@ private int get_long_log_sub_cc() {
 			}
 		}
 	}
-
-	private byte[] get_log_bytes(byte[] data_byte, int data_offset, int data_len) {
+	private byte[] get_unsigned_bytes(byte[] data_byte, int data_offset, int data_len) {  // #300
+		/*
+		 * return byte array with leading 0 byte followed by data bytes. This
+		 * array format is used to initialize BigInteger with logical unsigned
+		 * value.  
+		 * 2022-02-21 dsh issue 300 support unsigned and signed 2s compliment negative numbers with leading x'ff' vs x'00'
+		 */
+		byte[] new_byte = new byte[data_len + 1]; // default all 0x00
+		System.arraycopy(data_byte, data_offset, new_byte, 1, data_len); // RPI 411
+		return new_byte;
+	}
+	private byte[] get_signed_bytes(byte[] data_byte, int data_offset, int data_len) {  // #300
 		/*
 		 * return byte array with leading 0 byte followed by data bytes. This
 		 * array format is used to initialize BigInteger with logical unsigned
-		 * value.
+		 * value.  
+		 * 2022-02-21 dsh issue 300 fix to support 2s compliment negative numbers with leading x'ff' vs x'00'
 		 */
 		byte[] new_byte = new byte[data_len + 1];
+		if (data_byte[data_offset] < 0) {
+			new_byte[0] = (byte)0xff; // issue 300 support 2s compliment negative integers
+		}
 		System.arraycopy(data_byte, data_offset, new_byte, 1, data_len); // RPI
```

Comment:
Since data_byte[data_offset] is already signed, I think the code could be simplified to
```
    byte[] new_byte = new byte[data_len];
    System.arraycopy(data_byte, data_offset, new_byte, 0, data_len);
```